core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -531,7 +538,32 @@ private ExternalSearcherManager createSearcherManager(SearchFactory externalSear
         try {
             try {
                 final DirectoryReader directoryReader = ElasticsearchDirectoryReader.wrap(DirectoryReader.open(indexWriter), shardId);
-                internalSearcherManager = new SearcherManager(directoryReader, new SearcherFactory());
+                internalSearcherManager = new SearcherManager(directoryReader, new SearcherFactory() {
+                    @Override
+                    public IndexSearcher newSearcher(IndexReader reader, IndexReader previousReader) throws IOException {
+                        if (breakerService != null) {
+                            final CircuitBreaker breaker = breakerService.getBreaker(CircuitBreaker.ACCOUNTING);
+
+                            // get all new segment readers and account for their ram usage
+                            Set<SegmentReader> readers = reader.leaves().stream()
```

Comment:
I think we should use an ordinary loop here it's much easier to read or you add a utility method to `Lucene.java`?

Reply:
I'll switch this out for a regular loop